src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2192,41 +2236,49 @@ void func_808332F4(Player* this, PlayState* play) {
 
 /**
  * Get the appropriate Idle animation based on current `modelAnimType`.
- * This is used as the "primary" idle animation.
+ * This is the default idle animation.
  *
- * For special idle animations (which for example, change based on environment)
- * see `sSpecialIdleAnimations`.
+ * For fidget idle animations (which can for example, change based on environment)
+ * see `sFidgetAnimations`.
  */
 LinkAnimationHeader* Player_GetIdleAnim(Player* this) {
     return GET_PLAYER_ANIM(PLAYER_ANIMGROUP_wait, this->modelAnimType);
 }
 
 /**
- * Checks if the current animation is a "special" idle animation.
- * If it is, the index into `sSpecialIdleAnimations` is returned (plus one).
- * If the current animation is a "primary" idle animation, -1 is returned.
+ * Return values for `Player_CheckForIdleAnim`
+ */
+#define IDLE_ANIM_DEFAULT -1
+#define IDLE_ANIM_NONE 0
+// Fidget idle anims are returned by index. See `sFidgetAnimations` and `FidgetType`.
+
+/**
+ * Checks if the current animation is an idle animation.
+ * If the current animation is a fidget animation, the index into
+ * `sFidgetAnimations` is returned (plus one).
+ * If the current animation is a default idle animation, -1 is returned.
  * Lastly if the current animation is neither of these, 0 is returned.
  */
-s32 Player_CheckSpecialIdleAnim(Player* this) {
+s32 Player_CheckForIdleAnim(Player* this) {
     if (Player_GetIdleAnim(this) != this->skelAnime.animation) {
-        LinkAnimationHeader** specialAnim;
+        LinkAnimationHeader** fidgetAnim;
         s32 i;
 
-        for (i = 0, specialAnim = &sSpecialIdleAnimations[0][0]; i < 28; i++, specialAnim++) {
-            if (this->skelAnime.animation == *specialAnim) {
+        for (i = 0, fidgetAnim = &sFidgetAnimations[0][0]; i < 28; i++, fidgetAnim++) {
```

Comment:
```suggestion
        for (i = 0, fidgetAnim = &sFidgetAnimations[0][0]; i < FIDGET_MAX * ARRAY_COUNT(sFidgetAnimations[0]); i++, fidgetAnim++) {
```
Sorry, 1 extra comment (approval still applies). You could always add a fidget max to the enum to remove this constant here. Or could just do something like ` ARRAY_COUNT(sFidgetAnimations) * ARRAY_COUNT(sFidgetAnimations[0])`

Reply:
Ill go with max, sure

Reply:
though maybe we should consider a macro that does the second thing as a `ARRAY_COUNT_2D` or something, as a more general solution

Reply:
Opted to do this. Not sure if there are any more places where it can be used, but modders may find it useful anyway. I can think of a couple times where I've wanted it.

Reply:
how about `(s32)(sizeof(arr) / sizeof(arr[0][0]))`? not that it matters much

Reply:
sure

Reply:
I just copied what engineer suggested, didnt see its already in MM. I think yours makes sense and it can probably be ported over there if they agree